static/components/List.js

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react';
+import ListItem from './ListItem';
+import uuidv4 from 'uuid/v4'; // generate unique id
+
+class List extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            items: []
+        };
+        this.removeItem = this.removeItem.bind(this)
+    }
+
+    addTodo = (text) => {
+        var newItem = {
+            text: text,
+            date: new Date().toLocaleTimeString(),
+            id: uuidv4()
+        }
+        var newItems = [...this.state.items, newItem];
+
+        this.setState({
+            items: newItems
+        });
+    }
+
+    removeItem (item) {
```

Comment:
Modifying this method as "addTodo" method (to arrow functions) will give you opportunity to avoid context binding in constructor.  Moreover, we should aim similarity throughout the code.